app/components/map/legend/component.stories.tsx

```diff
@@ -27,14 +32,22 @@ const Template: Story<LegendProps> = (args) => {
   return (
     <Legend
       {...args}
+      maxHeight={300}
       onChangeOrder={onChangeOrder}
     >
-      {sortedItems.map((i) => (
-        <LegendItem
-          key={i.id}
-          {...i}
-        />
-      ))}
+      {sortedItems.map((i) => {
+        const { type, items } = i;
+        return (
+          <LegendItem
+            key={i.id}
+            {...i}
+          >
+            {type === 'basic' && <LegendTypeBasic className="text-sm text-gray-300" items={items} />}
+            {type === 'choropleth' && <LegendTypeChoropleth className="text-sm text-gray-300" items={items} />}
+            {type === 'gradient' && <LegendTypeGradient className="text-sm text-gray-300" items={items} />}
+          </LegendItem>
+        );
+      })}
```

Comment:
```
// pseudocode
{sortedItems.map(({type, items, id, ...rest}) => <LegendItem
            key={i.id} // wouldn't  type be better?
            {...rest}
          >
            {type === 'basic' && <LegendTypeBasic className="text-sm text-gray-300" items={items} />}
            {type === 'choropleth' && <LegendTypeChoropleth className="text-sm text-gray-300" items={items} />}
            {type === 'gradient' && <LegendTypeGradient className="text-sm text-gray-300" items={items} />}
          </LegendItem>
)}
```

Reply:
_**"wouldn't  type be better?"**_
Not at all. If you have legend items of the same type you will have the same keys for different items. Not desirable

Reply:
Ah indeed, I wrongly assumed that only those 3 will be present (i.e. it couldn't have multiple `basic` types)